### ISSUE TITLE
Parts previously generated previews not consistently working

### ISSUE THREAD

**Issue by JBEmbedded** [KittyCAD/text-to-cad-ui#82]
My previous generations don't alway show the part in the preview box. See pic attached! 
<img width="1087" alt="Screenshot 2023-12-23 at 11 30 28 AM" src="https://github.com/KittyCAD/text-to-cad-ui/assets/78819085/8fb4cab3-b61b-40d3-b9d4-db8c93e90ea6">

**Comment by alteous** [comment]
Are the parts regenerated every time or cached? I'm wondering whether it has anything to do with the glTF extension changing, although the date is fairly recent.

**Comment by franknoirot** [comment]
I think it's more likely something sillier with my UI @alteous but I'll keep an eye out for that, appreciate it.

**Comment by franknoirot** [comment]
Gotten some headway here with quick fixes around using a common derived store for both generations from prompt submissions and those fetched via a GET request, but there's a larger issue looming around maintaining enough canvas contexts. If I scroll to the bottom of my feed and back up again now I start to see breaking previews:

![Screenshot of text-to-cad-ui showing broken canvases at the top, with telltale errors in the console saying "Too many active WebGL contexts. Oldest context will be lost."](https://github.com/KittyCAD/text-to-cad-ui/assets/23481541/42baaeb9-0bce-4a7f-8f39-2241da9f19ba)

```
Too many active WebGL contexts. Oldest context will be lost.
```

Which led me to [this THREE.js form discussion](https://discourse.threejs.org/t/i-have-a-problem-in-warning-too-many-active-webgl-contexts-oldest-context-will-be-lost/41300/6) which points out that browsers really in principle only allow _one WebGL context per page_, although in practice depending on device and browser it could allow for a handful more.

I'm hoping that threlte has some config to help me clear the context for real between rendering different `<Canvas />` components (which I do already based on scroll position), so that I can prevent this behavior, but in summary for my own notes there are a few interlocking issues:
1. synchronizing the state of two lists of generations (submitted and fetched) is hard, and prone to being dropped from memory.
2. creating and destroying canvases, and by extension WebGL contexts, is unreliable
3. creating more than one canvas can be rejected depending on device and browser pair

After that rubber ducking, I'm going to try simply hoisting up the wrapper `<Canvas />` component from threlte to the app level, and just have one across all views, and see if that helps things a lot. It feels like that might have been an unstated assumption in their docs.

**Comment by franknoirot** [comment]
That approach didn't work, and led me down a little research run to [this very recent threlte issue I've commented on here](https://github.com/threlte/threlte/issues/753#issuecomment-1874683440). TLDR: React's THREEjs library has a clever `<View />` component that lets components share one canvas, and this threlte library needs to implement it, and I might have a go at it.

**Comment by franknoirot** [comment]
Yeah [this seems like a really promising example](https://github.com/pmndrs/react-three-next?tab=readme-ov-file), and [tunnel-rat](https://github.com/pmndrs/tunnel-rat) seems not too crazy to port to a Svelte implementation (it's for React).

**Comment by franknoirot** [comment]
Alright I'm still going to help threlte get their `<View />` component they've mostly built because I think that'll be valuable in the future, but we've decided this layout isn't necessary in this case, so I'm going to rework it to be more familiar to other chat AI tools and not need many canvasses on screen. I'll make an issue for that effort and reference this one.

**Comment by franknoirot** [comment]
Okay I believe this was resolved as a part of #92 but let me know if there are still any issues!
